crates/sui-core/src/checkpoints/reconstruction.rs

```diff
@@ -145,6 +145,10 @@ impl SpanGraph {
     ) -> (AuthorityName, usize) {
         let top1 = self.top_node(name1).0;
         let top2 = self.top_node(name2).0;
+        if top1 == top2 {
```

Comment:
That is correct, and exactly the same code I fix in the active checkpointing PR. We also have a test there for this condition.